frontend/src/app/main/main/main.component.ts

```diff
@@ -2,22 +2,37 @@ import { Component, OnInit } from '@angular/core';
 import { IAccount } from 'src/app/shared/models/account.model';
 import { AccountService } from 'src/app/account/services/account.service';
 import { TransactionService } from 'src/app/transaction/services/transaction.service';
+import { ITransaction } from 'src/app/shared/models/transaction.model';
+import { Observable } from 'rxjs';
 
 @Component({
   selector: 'app-main',
   templateUrl: './main.component.html',
   styleUrls: ['./main.component.scss'],
 })
 export class MainComponent implements OnInit {
-  public accounts!: IAccount[];
+  public accounts$!: Observable<IAccount[]>;
+  public transactions$!: Observable<ITransaction[]>;
+
   constructor(
     private accountsService: AccountService,
     public transactionsService: TransactionService
   ) {}
 
   public ngOnInit(): void {
-    this.accountsService.getAccounts().subscribe({
-      next: (data) => (this.accounts = data),
+    this.accounts$ = this.accountsService.getAccounts();
+    this.accountsService.getTransactionsObservable().subscribe({
```

Comment:
Add, please pipe(UntilDestroyed(this)) in all place where you subscribe on observables